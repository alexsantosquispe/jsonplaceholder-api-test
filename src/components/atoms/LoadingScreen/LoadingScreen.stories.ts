import type { Meta, StoryObj } from '@storybook/react';

import { LoadingScreen } from './LoadingScreen';

const meta = {
  title: 'Atoms/LoadingScreen',
  component: LoadingScreen,
  parameters: {
    layout: 'centered'
  }
} satisfies Meta<typeof LoadingScreen>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {}
};
