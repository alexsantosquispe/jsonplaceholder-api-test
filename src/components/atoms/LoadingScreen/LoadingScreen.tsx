import { twMerge } from 'tailwind-merge';

interface LoadingScreenProps {
  containerClassName?: string;
  dotClassName?: string;
}

const DOT_STYLE =
  'h-4 w-4 animate-[bounce_0.65s_ease-in-out_infinite] rounded-full bg-black/80 dark:bg-white/80';

export const LoadingScreen = ({
  containerClassName = '',
  dotClassName = ''
}: LoadingScreenProps) => {
  return (
    <div
      className={twMerge('flex h-screen w-full', containerClassName)}
      role="status"
      aria-label="Loading"
    >
      <div className="flex w-full items-center justify-center space-x-3">
        <span
          className={twMerge(
            DOT_STYLE,
            '[animation-delay:-0.4s]',
            dotClassName
          )}
        />
        <span
          className={twMerge(
            DOT_STYLE,
            '[animation-delay:-0.3s]',
            dotClassName
          )}
        />
        <span
          className={twMerge(
            DOT_STYLE,
            '[animation-delay:-0.2s]',
            dotClassName
          )}
        />
        <span
          className={twMerge(
            DOT_STYLE,
            '[animation-delay:-0.1s]',
            dotClassName
          )}
        />
        <span className={twMerge(DOT_STYLE, dotClassName)} />
      </div>
    </div>
  );
};
