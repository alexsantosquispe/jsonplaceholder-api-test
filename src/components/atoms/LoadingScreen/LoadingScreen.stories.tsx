import type { Meta, StoryObj } from '@storybook/react';

import { LoadingScreen } from './LoadingScreen';
import { Wrapper } from '../../../utils/stories-utils';

const meta = {
  title: 'Atoms/LoadingScreen',
  component: LoadingScreen,
  parameters: {
    layout: 'fullscreen'
  }
} satisfies Meta<typeof LoadingScreen>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = () => {
  return (
    <Wrapper className="w-full">
      <LoadingScreen />
    </Wrapper>
  );
};
Default.args = {};
