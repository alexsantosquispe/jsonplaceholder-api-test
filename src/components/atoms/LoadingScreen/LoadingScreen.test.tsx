import { render } from '@testing-library/react';

import { LoadingScreen } from './LoadingScreen';

describe('Loading Screen', () => {
  it('should render the component correctly', () => {
    const component = render(<LoadingScreen />);

    expect(component).toMatchSnapshot();
  });
});
