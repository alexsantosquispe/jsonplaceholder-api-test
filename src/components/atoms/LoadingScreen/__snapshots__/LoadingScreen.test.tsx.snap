// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Screen should render the component correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        aria-label="Loading"
        class="flex h-screen w-full"
        role="status"
      >
        <div
          class="flex w-full items-center justify-center space-x-3"
        >
          <span
            class="h-4 w-4 animate-[bounce_0.65s_ease-in-out_infinite] rounded-full bg-black/80 dark:bg-white/80 [animation-delay:-0.4s]"
          />
          <span
            class="h-4 w-4 animate-[bounce_0.65s_ease-in-out_infinite] rounded-full bg-black/80 dark:bg-white/80 [animation-delay:-0.3s]"
          />
          <span
            class="h-4 w-4 animate-[bounce_0.65s_ease-in-out_infinite] rounded-full bg-black/80 dark:bg-white/80 [animation-delay:-0.2s]"
          />
          <span
            class="h-4 w-4 animate-[bounce_0.65s_ease-in-out_infinite] rounded-full bg-black/80 dark:bg-white/80 [animation-delay:-0.1s]"
          />
          <span
            class="h-4 w-4 animate-[bounce_0.65s_ease-in-out_infinite] rounded-full bg-black/80 dark:bg-white/80"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      aria-label="Loading"
      class="flex h-screen w-full"
      role="status"
    >
      <div
        class="flex w-full items-center justify-center space-x-3"
      >
        <span
          class="h-4 w-4 animate-[bounce_0.65s_ease-in-out_infinite] rounded-full bg-black/80 dark:bg-white/80 [animation-delay:-0.4s]"
        />
        <span
          class="h-4 w-4 animate-[bounce_0.65s_ease-in-out_infinite] rounded-full bg-black/80 dark:bg-white/80 [animation-delay:-0.3s]"
        />
        <span
          class="h-4 w-4 animate-[bounce_0.65s_ease-in-out_infinite] rounded-full bg-black/80 dark:bg-white/80 [animation-delay:-0.2s]"
        />
        <span
          class="h-4 w-4 animate-[bounce_0.65s_ease-in-out_infinite] rounded-full bg-black/80 dark:bg-white/80 [animation-delay:-0.1s]"
        />
        <span
          class="h-4 w-4 animate-[bounce_0.65s_ease-in-out_infinite] rounded-full bg-black/80 dark:bg-white/80"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
